src/acquisition/covidcast/direction_updater.py

```diff
@@ -278,21 +302,22 @@ def main(
 
   `args`: parsed command-line arguments
   """
-
-  database = database_impl()
-  database.connect()
-  commit = False
-
-  try:
-    update_loop_impl(database)
-    # only commit on success so that directions are consistent with respect
-    # to methodology
-    commit = True
-  finally:
-    # no catch block so that an exception above will cause the program to
-    # fail after the following cleanup
-    database.disconnect(commit)
-    print('committed=%s' % str(commit))
+  for partition_index in args.partitions:
+    database = database_impl()
+    database.connect()
+    commit = False
+    try:
+      update_loop_impl(database, partition_index)
+      # only commit on success so that directions are consistent with respect
+      # to methodology
+      commit = True
+    except Exception as e:
+      raise e
+    finally:
+      # no catch block so that an exception above will cause the program to
+      # fail after the following cleanup
+      database.disconnect(commit)
```

Comment:
possibly consider moving this out of the loop so the db is consistent?  i guess having partially updated directions isnt such a bad thing

Reply:
I guess there is an argument for both sides. The other side being this allows to investigate the failure and only update the remaining partitions instead of running the entire job again.

Reply:
¯\_(ツ)_/¯ its probably fine to leave it

Reply:
Either way someone has to go in and fix it manually, so as long as we document what the behavior is and what should be done in a failure, it's fine.